### PR TITLE
GLOB-50762 upgrade more dependencies

### PR DESCRIPTION
**Pull request by simpleigh-globality**
Closes [GLOB-50762]

### `yarn outdated`

**WAS:**

```
Package                               Current Wanted  Latest  Package Type    URL                                                                            
@babel/cli                            7.7.7   7.12.10 7.12.10 devDependencies https://babeljs.io/                                                            
@babel/core                           7.7.7   7.12.10 7.12.10 devDependencies https://babeljs.io/                                                            
@babel/node                           7.7.7   7.12.10 7.12.10 devDependencies https://babeljs.io/                                                            
@babel/preset-env                     7.7.7   7.12.11 7.12.11 devDependencies https://babeljs.io/                                                            
@globality/nodule-express             0.1.14  0.1.14  0.2.0   devDependencies https://github.com/globality-corp/nodule-express#readme                        
@globality/nodule-openapi             0.17.2  0.17.2  0.19.1  devDependencies https://github.com/globality-corp/nodule-openapi#readme                        
apollo-server-express                 2.9.14  2.19.2  2.19.2  dependencies    https://github.com/apollographql/apollo-server#readme                          
axios                                 0.18.1  0.18.1  0.21.1  dependencies    https://github.com/axios/axios                                                 
dataloader                            1.4.0   1.4.0   2.0.0   dependencies    https://github.com/graphql/dataloader                                          
enum                                  2.5.0   2.5.0   3.0.4   dependencies    https://github.com/adrai/enum                                                  
eslint                                4.19.1  4.19.1  7.18.0  devDependencies https://eslint.org                                                             
eslint-config-airbnb-base             12.1.0  12.1.0  14.2.1  devDependencies https://github.com/airbnb/javascript                                           
eslint-plugin-import                  2.19.1  2.22.1  2.22.1  devDependencies https://github.com/benmosher/eslint-plugin-import                              
graphql                               14.5.8  14.7.0  15.4.0  devDependencies https://github.com/graphql/graphql-js                                          
graphql-playground-middleware-express 1.7.12  1.7.22  1.7.22  dependencies    https://github.com/graphcool/graphql-playground/tree/master/packages/graphql-playground-middleware-express
helmet                                3.21.2  3.23.3  4.4.1   dependencies    https://helmetjs.github.io/                                                    
http-status-codes                     1.4.0   1.4.0   2.1.4   dependencies    https://github.com/prettymuchbryce/node-http-status#readme                     
jest                                  24.9.0  24.9.0  26.6.3  devDependencies https://jestjs.io/                                                             
lodash                                4.17.15 4.17.20 4.17.20 dependencies    https://lodash.com/                                                            
supertest                             3.4.2   3.4.2   6.1.2   devDependencies https://github.com/visionmedia/supertest#readme                                
throat                                4.1.0   4.1.0   6.0.1   dependencies    https://github.com/ForbesLindesay/throat#readme                                
uuid                                  3.3.3   3.4.0   8.3.2   dependencies    https://github.com/uuidjs/uuid#readme
```

**NOW:**

```
Package                   Current Wanted Latest Package Type    URL
@globality/nodule-express 0.1.14  0.1.14 0.2.0  devDependencies https://github.com/globality-corp/nodule-express#readme
dataloader                1.4.0   1.4.0  2.0.0  dependencies    https://github.com/graphql/dataloader
enum                      2.5.0   2.5.0  3.0.4  dependencies    https://github.com/adrai/enum
graphql                   14.7.0  14.7.0 15.4.0 devDependencies https://github.com/graphql/graphql-js
helmet                    3.23.3  3.23.3 4.4.1  dependencies    https://helmetjs.github.io/
```

### Transpiled library

I diffed the transpiled source between `master` and this branch. The full diff includes many changes to comments, but here's the important bits:

```patch
diff --recursive lib-old/index.js lib-new/index.js
102,107d101
< Object.defineProperty(exports, "setCache", {
<   enumerable: true,
<   get: function () {
<     return _services.setCache;
<   }
< });
diff --recursive lib-old/middleware/jwt/errors.js lib-new/middleware/jwt/errors.js
37c38
<   return addHeader(req, res.status(_httpStatusCodes.UNAUTHORIZED), realm).json({
---
>   return addHeader(req, res.status(_httpStatusCodes.StatusCodes.UNAUTHORIZED), realm).json({
diff --recursive lib-old/middleware/jwt/middleware.js lib-new/middleware/jwt/middleware.js
69c69
<           return res.status(_httpStatusCodes.UNAUTHORIZED).end();
---
>           return res.status(_httpStatusCodes.StatusCodes.UNAUTHORIZED).end();
78c78
<           return res.status(_httpStatusCodes.UNAUTHORIZED).end();
---
>           return res.status(_httpStatusCodes.StatusCodes.UNAUTHORIZED).end();
106c106
<             return res.status(_httpStatusCodes.UNAUTHORIZED).end();
---
>             return res.status(_httpStatusCodes.StatusCodes.UNAUTHORIZED).end();
109c109
<             return res.status(_httpStatusCodes.UNAUTHORIZED).end();
---
>             return res.status(_httpStatusCodes.StatusCodes.UNAUTHORIZED).end();
diff --recursive lib-old/services/core/keys.js lib-new/services/core/keys.js
10c10
< var _v = _interopRequireDefault(require("uuid/v5"));
---
> var _uuid = require("uuid");
14,15d13
< function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
< 
36c35
<   const namespace = (0, _lodash.get)((0, _noduleConfig.getContainer)('config.cache'), 'namespace', _v.default.URL);
---
>   const namespace = (0, _lodash.get)((0, _noduleConfig.getContainer)('config.cache'), 'namespace', _uuid.v5.URL);
39c38
<   return (0, _v.default)(keyString, namespace);
---
>   return (0, _uuid.v5)(keyString, namespace);
diff --recursive lib-old/services/dedup/wrapper.js lib-new/services/dedup/wrapper.js
11,12d10
< var _v = _interopRequireDefault(require("uuid/v4"));
< 
14a13,14
> var _uuid = require("uuid");
> 
74c77
<   const loaderId = loaderName || (0, _v.default)(); // Fetch the loader in call time - (and in the req init)
---
>   const loaderId = loaderName || (0, _uuid.v4)(); // Fetch the loader in call time - (and in the req init)
```

The changes here are largely explained by:

* removal of `setCache` export
* upgrade to `http-status-codes`
* upgrade to `uuid`

[GLOB-50762]: https://globality.atlassian.net/browse/GLOB-50762

### CODE REVIEW

**Review by german-glob — DISMISSED** [review]

**Comment by simpleigh-globality** [comment]
Whoops, sorry, should have created this as a draft - still in progress 😞

File: package.json [hunk 1]
```diff
@@ -16,48 +16,46 @@
         "preyalc": "yarn clean && yarn build"
     },
     "dependencies": {
-        "apollo-server-express": "^2.9.9",
-        "axios": "^0.18.0",
-        "connect-requestid": "^1.1.0",
-        "cookie-parser": "^1.4.5",
```
Comment (simpleigh-globality) [inline package.json:22]: This isn't imported anywhere so I removed it.

File: package.json [hunk 2]
```diff
@@ -16,48 +16,46 @@
...
-        "express": "^4.16.3",
-        "express-jwt": "^6.0.0",
-        "graphql-playground-middleware-express": "^1.7.12",
-        "helmet": "^3.12.0",
-        "http-status-codes": "^1.3.0",
-        "is-uuid": "^1.0.2",
-        "jsonwebtoken": "^8.2.0",
-        "lodash": "^4.17.5",
-        "spookyhash": "^1.2.0",
-        "throat": "^4.1.0",
```
Comment (simpleigh-globality) [inline package.json:35]: This isn't imported anywhere so I removed it.

File: package.json [hunk 3]
```diff
@@ -16,48 +16,46 @@
...
+        "axios": "0.21.1",
+        "connect-requestid": "1.1.0",
+        "cors": "2.8.5",
+        "dataloader": "1.4.0",
+        "enum": "2.5.0",
+        "express": "4.17.1",
+        "express-jwt": "6.0.0",
+        "graphql-playground-middleware-express": "1.7.22",
+        "helmet": "3.23.3",
+        "http-status-codes": "2.1.4",
```
Comment (simpleigh-globality) [inline package.json:29]: Manual major version upgrade.

File: package.json [hunk 4]
```diff
@@ -16,48 +16,46 @@
...
+        "is-uuid": "1.0.2",
+        "jsonwebtoken": "8.5.1",
+        "lodash": "4.17.20",
+        "spookyhash": "1.2.0",
+        "uuid": "8.3.2"
```
Comment (simpleigh-globality) [inline package.json:34]: Manual major version upgrade.

File: src/enums/index.js [hunk 5]
```diff
@@ -3,15 +3,14 @@
 import { GraphQLEnumType } from 'graphql';
 
 // converts an Enum type created using the 'enum' package to a GraphQLEnumType
-const createGraphQLEnumType = (standardEnum, name, description = '') =>
```
Comment (simpleigh-globality) [inline src/enums/index.js:6]: ESLint no longer allows a line break after an arrow function, which is annoying.

File: src/index.js [hunk 6]
```diff
@@ -30,5 +30,4 @@ export {
     ANY_SINGLE_ITEM_LIST,
     ANY_UUID,
     CachingSpec,
-    setCache,
```
Comment (simpleigh-globality) [inline src/index.js:33]: This doesn't exist in `./services`. It doesn't exist anywhere in the library.

**Review by german-glob — APPROVED** [review]

**Review by alex-roe-globality — APPROVED** [review]

**Review by ntnsch — APPROVED** [review]

**Pull request merged**